science/eso-flames-fposs/Portfile

```
@@ -0,0 +1,87 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+
+#PortGroup       compilers 1.0
+
+set instrument      flames
+
+name            eso-flames-fposs
+
+version         115.0.3
+revision        2
+
+categories      science
+license         GPL-2+
+maintainers     eso.org:john.pritchard
+homepage        http://www.eso.org/sci/observing/phase2/SMGuidelines/FPOSS.FLAMES.html
+description     ESO FLAMES Observation Preparation software
+long_description ESO FLAMES Observation Preparation software.
+                 
+master_sites        https://ftp.eso.org/pub/usg/fposs/
+distname            flames-fposs-${version}
+distfiles           fposs-${version}.tar.gz
+
+checksums           fposs-115.0.3.tar.gz \
+                    rmd160  065a65f558317661d1356bfeee579f8790c14ecb  \
+                    sha256  9d16463dacfaa493a2221c352b2aa323bd0edd2f86c3d1fcafe6588a11f0989b  \
+                    size    2319171
+
+set worksrcpath          ${workpath}/fposs-${version}
+
+#compilers.choose    cc cxx
+#compilers.setup     gcc
+
+configure.cflags-append    -std=c11
+configure.cxxflags-append  -std=c11
+
+#default_variants  +${compilers.gcc_default}
+use_parallel_build      no
+
+depends_run       port:tk port:tcl \
+                  port:ncurses \
+                  port:xorg-server
+
+#variant xquartz   description {Use this variant if you have XQuartz installed.} {
+#  depends_run-remove port:xorg-server
+#  depends_run-append port:xorg-libX11
+#}
+
+configure {
+  reinplace "s@\$(CC) \$(CFLAGS) -c \$(PAFCHK_MODULE)@\$(CC) -DFLAMES_ONLY -DDRAMA_BUILD \$(CWARNINGS) \$(COPT) -I\$(PAFCHK_MODULE)/include -c \$(PAFCHK_MODULE)@" \
```

Using reinplace like this is permissible, but be aware that it can silently fail and give you a wrong result when the upstream source changes. For this reason we tend to prefer using a patch file, since that will unambiguously fail if it doesn't apply cleanly.

When you don't want variable or command substitution inside a string, you can quote it in braces and avoid having to use a bunch of backslashes.

---

All replaces moved to patch files.

---

Unfortunately it's not possible to use a patch alone when there are Portfile variables that need to be substituted. But the patch itself can set a placeholder value which can then be reliably changed with reinplace. An example of this is given here: https://trac.macports.org/wiki/PortfileRecipes#hardcode_opt_local

It looks like this needs to be done here for `CC` and `prefix`/`eso.prefix`.

---

Ah true! I was a bit too hasty! Thanks for catching that. I've reworked the patches to set `@@ESO_CC@@` and `@@ESO_PREFIX@@` placeholders and then `reinplace` those in the port file.